src/IO/WriteHelpers.h

```diff
@@ -765,7 +765,7 @@ template <char date_delimeter = '-', char time_delimeter = ':', char between_dat
 inline void writeDateTimeTextWithTimeZone(time_t datetime, WriteBuffer & buf, const DateLUTImpl & time_zone = DateLUT::instance())
 {
     writeDateTimeText<date_delimeter, time_delimeter, between_date_time_delimiter>(datetime, buf,  time_zone);
-    auto offset = time_zone.timezoneOffset(datetime);
+    auto offset = time_zone.timezoneOffset(std::abs(datetime));
     auto asb_offset = std::abs(offset);
     auto hours = asb_offset / 3600;
     auto minutes = asb_offset % 3600 / 60;
```

Comment:
need to test some timezones with `negative offset`, such as `2024-01-01 00:00:00-01:00`

Reply:
There are some existed cases and the performance is correct
![image](https://github.com/user-attachments/assets/028bd414-9929-49d9-aec2-23ea72dce484)